xbmc/windows/GUIMediaWindow.cpp

```diff
@@ -1455,10 +1454,20 @@ void CGUIMediaWindow::OnInitWindow()
 
   m_rootDir.SetAllowThreads(true);
 
-  if (m_iSelectedItem > -1)
-    m_viewControl.SetSelectedItem(m_iSelectedItem);
-
   CGUIWindow::OnInitWindow();
+
+  m_viewControl.SetSelectedItem(GetMediaSelectedItem(m_vecItems->GetPath()));
+}
+
+int CGUIMediaWindow::GetMediaSelectedItem(const std::string& strDirectory)
+{
+  auto it = m_mediaSelectedItems.find(strDirectory);
+  return (it != m_mediaSelectedItems.end()) ? it->second : 0;
```

Comment:
at the cost of being repetitive let me explain better why this patch

without these changes in CGUIMediaWindow on GUI_MSG_WINDOW_DEINIT will be stored the selected item in the integer m_iSelectedItem, then on OnInitWindow() will be used this value to reselect the last selected item

the problem is that the instance is the same for the sections videos, tvshows and movies, so if you select the item 10 in tv shows and then you go back to the main menu, when you enter in the section videos or movies it will be selected the item 10 (and viceversa)

to avoid this beahvior I replaced m_iSelectedItem with m_mediaSelectedItems to save also the current path (tvshows/videos/movies/) with the last selected item

to answer your question in GetMediaSelectedItem will be sent the current path, if we already entered in that particular section and so m_mediaSelectedItems.find(strDirectory) find a match, will be returned the integer value of the item associated to that path (it->second) otherswise will be returned 0 and the gui will select the first element on top